### PR TITLE
get rename to work on new panels, need to investigate old panels

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
fix rename flow on old panels
fix rename flows on the individual panels page
fix delete flow (still need to navigate back if success delete on SO panels (works for old panels))

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/custom_panels/custom_panel_view_so.tsx [hunk 1]
```diff
@@ -218,10 +210,14 @@ export const CustomPanelViewSO = (props: CustomPanelViewProps) => {
...
 
-  const deletePanel = () => {
+  const deletePanelInteraction = () => {
     setModalLayout(
       <DeleteModal
-        onConfirm={onDelete}
+        onConfirm={() => {
+          dispatch(deletePanel(panelId));
+          closeModal();
+          // TODO : navigate back if successful
```
Comment (derek-ho) [inline public/components/custom_panels/custom_panel_view_so.tsx:219]: needs to be done @pjfitzgibbons

File: public/components/custom_panels/redux/panel_slice.ts [hunk 2]
```diff
@@ -131,7 +131,7 @@ const updateSavedObjectPanel = (panel: CustomPanelType) => savedObjectPanelsClie
 
 export const uuidRx = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/;
 
-const isUuid = (id) => !!id.match(uuidRx);
```
Comment (derek-ho) [inline public/components/custom_panels/redux/panel_slice.ts:134]: is this ok?

File: public/components/custom_panels/redux/panel_slice.ts [hunk 3]
```diff
@@ -242,7 +242,7 @@ export const renameCustomPanel = (editedCustomPanelName: string, id: string) =>
   }
 
   const panel = getState().customPanel.panelList.find((p) => p.id === id);
-  const updatedPanel = { ...panel, title: editedCustomPanelName };
+  const updatedPanel = { ...panel, title: editedCustomPanelName, name: editedCustomPanelName };
```
Comment (derek-ho) [inline public/components/custom_panels/redux/panel_slice.ts:245]: to get rename old panels to work

File: public/components/custom_panels/custom_panel_view.tsx [hunk 4]
```diff
@@ -30,6 +30,7 @@ import React, { useEffect, useState } from 'react';
 import { DurationRange } from '@elastic/eui/src/components/date_picker/types';
 import moment from 'moment';
 import _ from 'lodash';
+import { useDispatch, useSelector } from 'react-redux';
```
Comment (derek-ho) [inline public/components/custom_panels/custom_panel_view.tsx:33]: some tests are failing due to not being wrapped in a provider

File: public/components/custom_panels/redux/panel_slice.ts (hunk 2 above)
Comment (joshuali925) [reply public/components/custom_panels/redux/panel_slice.ts:134]: use `uuidRx.test(id)`

**Comment by derek-ho** [comment]
> Can you explain or point-to how this change fixes rename? I see changes to the redux-stored `panel`, I don't understand why this change to rename worked?

It fixes it because under the hood observability panel uses name instead of title - so we need to save that in updatepanel in case its old panel

**Comment by anirudha** [comment]
dupe

**Pull request closed without merging**